libs/providers/go-feature-flag/src/lib/data-collector-hook.ts

```diff
@@ -100,7 +103,7 @@ export class GoFeatureFlagDataCollectorHook implements Hook {
     evaluationDetails: EvaluationDetails<FlagValue>,
     hookHints?: HookHints
   ) {
-    if (evaluationDetails.reason !== StandardResolutionReasons.CACHED){
+    if (!this.collectUnCachedEvaluation && evaluationDetails.reason !== StandardResolutionReasons.CACHED){
```

Comment:
I'm having trouble understanding the intent here. Should this be an or? What's the expected behavior?

Reply:
Ok, I think I understand.